node_modules/helpers/modules/memdb/index.js

```diff
@@ -29,10 +30,44 @@ function read(filename) {
     }
 }
 
+function findInHMap(input, storageName) {
+    const index = hash(input);
+
+    if(!cachedb[storageName][index]) {
+        throw new Error('Object is not found.');
+    }
+
+    return cachedb[storageName][index];  
+}
+
+function addInHMap(input, storageName, data) {
+    const index = hash(input);
+
+    if(cachedb[storageName] == null) {
+        cachedb[storageName] = {};
+    }
+
+    if(cachedb[storageName][index] == null) {
+        cachedb[storageName][index] = [];
+    }
+
+    for(const objects of cachedb[storageName][index]) {
+        if(objects == data) { 
+            throw new Error('Object is already in the hashmap.');
+        }
+    }
+
+    cachedb[storageName][index].push(data);
+}
+
 module.exports = {
     memdb: {
         write: write,
         read: read,
-        cache: cachedb
+        cache: cachedb,
+        map: {
```

Comment:
For better clarity I think we should rename `map` to `hash` and it's corresponding methods should be `find` and `map`. Since both our functions make reference to hashing and because we make use of the "hash.js" dependency it makes sense that our exported objects would too :)
So we would have:
```js
hash: {
   find: findInHMap,
   map: addInHMap
}
```